From a439ec05e6eca66813102f491b8e42f16c7f213e Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@ariadne.space>
Date: Sun, 12 Oct 2025 00:54:52 -0700
Subject: [PATCH] Revert "s390x: manually inline __tls_get_addr in
 __tls_get_offset"

This reverts commit 6af4f25b899e89e4b91f8c197ae5a6ce04bcce7b.
---
 src/thread/s390x/__tls_get_offset.s | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/thread/s390x/__tls_get_offset.s b/src/thread/s390x/__tls_get_offset.s
index 405f118b..8ee92de8 100644
--- a/src/thread/s390x/__tls_get_offset.s
+++ b/src/thread/s390x/__tls_get_offset.s
@@ -1,17 +1,17 @@
 	.global __tls_get_offset
 	.type __tls_get_offset,%function
 __tls_get_offset:
-	ear   %r0, %a0
-	sllg  %r0, %r0, 32
-	ear   %r0, %a1
+	stmg  %r14, %r15, 112(%r15)
+	aghi  %r15, -160
 
-	la    %r1, 0(%r2, %r12)
+	la    %r2, 0(%r2, %r12)
+	brasl %r14, __tls_get_addr
 
-	lg    %r3, 0(%r1)
-	sllg  %r4, %r3, 3
-	lg    %r5, 8(%r0)
-	lg    %r2, 0(%r4, %r5)
-	ag    %r2, 8(%r1)
-	sgr   %r2, %r0
+	ear   %r1, %a0
+	sllg  %r1, %r1, 32
+	ear   %r1, %a1
 
+	sgr   %r2, %r1
+
+	lmg   %r14, %r15, 272(%r15)
 	br    %r14
-- 
2.51.0

